fi/include/reverse_purge_hash_map.hpp

```diff
@@ -91,8 +91,14 @@ class reverse_purge_hash_map {
 
 // This iterator uses strides based on golden ratio to avoid clustering during merge
 template<typename K, typename V, typename H, typename E, typename A>
-class reverse_purge_hash_map<K, V, H, E, A>::iterator: public std::iterator<std::input_iterator_tag, K> {
+class reverse_purge_hash_map<K, V, H, E, A>::iterator {
 public:
+  using iterator_category = std::input_iterator_tag;
+  using value_type = std::pair<K&, V>;
+  using difference_type = std::ptrdiff_t;
```

Comment:
I think this can be void in this case. I would be much more comfortable not specifying pointer difference for temporary values

Reply:
Good point